### DISCUSSION TITLE
How to delete old builds

### DISCUSSION THREAD

**Discussion by slavus** [woodpecker-ci/woodpecker#2781] [Ask a Question]
Until this feature is implemented, how can we manually delete old builds? Which tables should be deleted? 
Does anybody have SQL I could run that would delete all builds and their logs beside last 10 builds

**Comment by anbraten** [comment]
There is an api endpoint for it AFAIK. Might even be included in the cli already.

**Reply by slavus** [reply]
There is a cli command to purge log. But only one specific log. My question is abaut maintance and to delete not only logs but also old builds. As far I can see there is no API to delete pipeline or to see which pipelines have deleted logs.

**Comment by qwerty287** [comment]
There is an API in `next`: https://github.com/woodpecker-ci/woodpecker/pull/3506
